src/platform/CMakeLists.txt

```diff
@@ -54,16 +66,96 @@ if("${CX_PLATFORM}" STREQUAL "windows")
     target_link_libraries(
         platform
         PUBLIC
-        inc
         wbemuuid)
-else()
-    target_link_libraries(platform PUBLIC inc)
-endif()
-
-if("${CX_PLATFORM}" STREQUAL "windows")
     target_link_libraries(platform PUBLIC winmm)
+elseif(LINUX_XDP_ENABLED)
+    find_library(NL_LIB nl-3)
+    find_library(NL_ROUTE_LIB nl-route-3)
+    if (NOT QUIC_BUILD_LIBXDP_FROM_SOURCE)
+        target_include_directories(platform PRIVATE /usr/include/xdp)
+        target_include_directories(platform PRIVATE /usr/include/bpf)
+        set(XDP_PROG_INCLUDE_DIR "-I/usr/include/bpf")
+
+        find_library(XDP_LIB libxdp.so)
+        find_library(BPF_LIB libbpf.so)
+    else ()
+        # XDP Tools and libbpf paths
+        set(XDP_TOOLS_DIR ${PROJECT_SOURCE_DIR}/submodules/xdp-tools)
```

Comment:
Do we only need the submodule if we support build from source?

Reply:
If so, I'd prefer not to need another submodule.

Reply:
I thought there is no strong reason to delete.
Enabling xdp from source required huge effort. People from other communities (xdp/ebpf etc.) can easily involve with this option.

Reply:
If we need the submodule to compile/link against the library that's one thing, but if this is here just to build the source, I prefer to not have that, at least not right now, in this PR. You can move the code to a follow up PR to save it.

Time has shown that keeping submodules and their builds up to date has been painful with openssl. We want as few of these dependencies as possible.